### PR TITLE
Enhance/module/hadoop

### PR DESCRIPTION
**Pull request by daxroc**
### What does this PR do?
Backward compatible Enhancement for hadoop module to use hdfs tool over deprecated hadoop

### New Behavior
Uses hdfs tool to perform the same commands enabled with optional arguments as described in the documentation

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/modules/hadoop.py [hunk 1]
```diff
@@ -63,23 +65,46 @@ def _hadoop_cmd(module, command, *args):
     return out
 
 
+def _hdfs_cmd(module, command, *args):
+  '''
+     hdfs command wrapper
+  '''
+  out = None
+  if module and command:
+    if module in __authorized_modules__:
```
Comment (s0undt3ch) [inline salt/modules/hadoop.py]: Are these tabs? Indentation should be a multiple of 4 spaces and no tabs.

**Review by s0undt3ch — COMMENTED** [review]
Is hdfs the new CLI binary? An alternative binary? We could probably search `$PATH` for the hdfs binary and then the dfs binary in this order since the new one is backwards compatible with the old one?
This would remove the need to explicitly pass `use_hdfs`?

These are basically questions to see if we can support both more transparently....

**Comment by daxroc** [comment]
@s0undt3ch made the adjustments to transparently use hdfs if available and fallback on hadoop if not. Refactored the version to use the same command wrapper. And some minor changes on lint recommendations.

Thanks for the feedback - appreciated!

File: salt/modules/hadoop.py [hunk 2]
```diff
@@ -14,47 +14,40 @@
...
 
 
 def __virtual__():
     '''
     Check if hadoop is present, then load the module
     '''
-    if salt.utils.which('hadoop'):
+    if salt.utils.which('hadoop') or salt.utils.which('hdfs'):
         return 'hadoop'
     return (False, 'The hadoop execution module cannot be loaded: hadoop binary not in path.')
```
Comment (s0undt3ch) [inline salt/modules/hadoop.py]: _hadoop or hdfs not found_

File: salt/modules/hadoop.py [hunk 3]
```diff
@@ -63,10 +56,27 @@ def _hadoop_cmd(module, command, *args):
...
+    module = 'version'
+    out = _hadoop_cmd(module, True).splitlines()[0].split()
+    return out[1]
+
+
 def dfs(command=None, *args):
     '''
     Execute a command on DFS
 
+    trailing argument of use_hdfs will use hdfs command for execution
```
Comment (s0undt3ch) [inline salt/modules/hadoop.py]: The docstring here is still valid?

**Review by cachedout — CHANGES_REQUESTED** [review]
A test has failed. Could you please take a look?

https://jenkins.saltstack.com/job/PR/job/salt-pr-rs-cent7-n/10295/testReport/junit/unit.modules.test_hadoop/HadoopTestCase/test_version/

**Comment by daxroc** [comment]
Reverted version method back - worked as it should have. My bad ...

**Review by cachedout — APPROVED** [review]

**Pull request merged**